src/scripts/cronjobs/monthlyActivityFree.tsx

```diff
@@ -0,0 +1,151 @@
+   * recommendations for outside the US).
+   */
+  const countryCodeGuess = getSignupLocaleCountry(subscriber);
+
+  // OneRep suggested not relying on webhooks, but instead to fetch the latest
+  // data from their API. Thus, let's refresh the data in our DB in real-time:
+  if (subscriber.onerep_profile_id !== null) {
+    await refreshStoredScanResults(subscriber.onerep_profile_id);
+  }
+
+  const latestScan = await getLatestOnerepScanResults(
+    subscriber.onerep_profile_id,
+  );
+  const subscriberBreaches = await getSubscriberBreaches({
+    fxaUid: subscriber.fxa_uid,
+    countryCode: countryCodeGuess,
+  });
+  const data = getDashboardSummary(latestScan.results, subscriberBreaches);
+
+  const subject = l10n.getString("email-monthly-free-subject");
+  const unsubscribeLink =
+    await getMonthlyActivityFreeUnsubscribeLink(subscriber);
+
+  if (unsubscribeLink === null) {
+    throw new Error(
+      `Trying to send a monthly activity email to a free user, but the unsubscribe link could not be generated: [${unsubscribeLink}].`,
+    );
+  }
+
+  // Update the last-sent date *first*, so that if something goes wrong, we
+  // don't keep resending the email a brazillion times.
+  await updateEmailPreferenceForSubscriber(subscriber.id, true, {
+    monthly_monitor_report_free_at: new Date(Date.now()),
+  });
+
+  try {
+    await sendEmail(
+      sanitizedSubscriber.primary_email,
+      subject,
+      renderEmail(
+        <MonthlyActivityFreeEmail
+          subscriber={sanitizedSubscriber}
+          dataSummary={data}
+          l10n={l10n}
+          unsubscribeLink={unsubscribeLink}
+        />,
+      ),
+    );
+  } catch (e) {
+    console.error("send_monthly_activity_email_free", {
+      exception: e,
+    });
+  }
+}
+
+export async function getMonthlyActivityFreeUnsubscribeLink(
+  subscriber: SubscriberRow,
+) {
+  try {
+    const newUnsubToken = randomBytes(64).toString("hex");
+    let sub;
+    const getRes = await getEmailPreferenceForSubscriber(subscriber.id);
+    if (getRes.unsubscribe_token) {
+      // if record has been created and the token exists, return the token
+      return `${process.env.SERVER_URL}/unsubscribe-email/monthly-report-free?token=${getRes.unsubscribe_token}`;
+    } else if (
+      !getRes.monthly_monitor_report_free_at &&
+      !getRes.unsubscribe_token
+    ) {
+      // if record in the new table has not been created
+      sub = await addUnsubscribeTokenForSubscriber(
+        subscriber.id,
+        newUnsubToken,
+      );
+    } else {
+      // if record already exists, but token doesn't exist, add the token
+      sub = await updateEmailPreferenceForSubscriber(subscriber.id, true, {
+        unsubscribe_token: newUnsubToken,
+      });
+    }
+
+    return `${process.env.SERVER_URL}/unsubscribe-email/monthly-report-free?token=${sub.unsubscribe_token}`;
+  } catch (e) {
+    console.error("generate_unsubscribe_link", {
+      exception: e as string,
+    });
```

Comment:
Same here re: logger (this uses [Winston](https://github.com/winstonjs/winston) and it understands [structured logging](https://cloud.google.com/logging/docs/structured-logging))

Reply:
Changed in f65d38b9b.